### ISSUE TITLE
[Bug] [UX] Change list scrolls to top when discarding & scrolls horizontally when cursor hovers scrollbar

### ISSUE THREAD

**Issue by KaKi87** [Murmele/Gittyup#514]
Hello,

Here are 2 UX problems related to the change list.

## It scrolls to top when discarding changes

Example case : there are 3+ changes, the first one is to be committed, the second one is to be discarded.

Reproduction steps :
1. Mark the first change to be committed ;
2. Scroll down to the second change ;
3. Discard it.

Expected behavior :
- the scroll position is preserved after the second change is discarded ;
- the third change takes the second's place, allowing it to processed without further scrolling.

Actual behavior :
- the scroll position is reset to the very top after the second change is discarded ;
- the user must scroll down through the first change again to continue processing the following ones.

## It scrolls horizontally when cursor hovers scrollbar

Example case : there are 2+ changes, the first one has horizontal scroll width.

Reproduction steps :
1. Position the cursor anywhere in the change list view ;
2. Scroll down ;
3. The cursor eventually hovers the first change's horizontal scrollbar ;
4. Continue scrolling.

Expected behavior : the change list continues to be scrolled down.

Actual behavior : the first change begins scrolling right.

---

Thanks

**Comment by jensenr30** [comment]
Addressing item number 2 (horizontal scroll UX issue):

@Murmele 

What do you think about implementing something like a single scroll bar at the bottom of the diff window that would simultaneously scroll all the diffs of each file in sync?  I have no clue how much work it would be to do this.

Concept:

![image](https://github.com/Murmele/Gittyup/assets/3083835/ee89f7ab-ad8a-471c-92b1-c4f0656374af)

**Comment by Murmele** [comment]
> Addressing item number 2 (horizontal scroll UX issue):
> 
> @Murmele
> 
> What do you think about implementing something like a single scroll bar at the bottom of the diff window that would simultaneously scroll all the diffs of each file in sync? I have no clue how much work it would be to do this.
> 
> Concept:
> 
> ![image](https://private-user-images.githubusercontent.com/3083835/307660047-ee89f7ab-ad8a-471c-92b1-c4f0656374af.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDg5MjkxNjYsIm5iZiI6MTcwODkyODg2NiwicGF0aCI6Ii8zMDgzODM1LzMwNzY2MDA0Ny1lZTg5ZjdhYi1hZDhhLTQ3MWMtOTJiMS1jNGYwNjU2Mzc0YWYucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDIyNiUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDAyMjZUMDYyNzQ2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzdlN2FmZTFjY2QzZjA1ZGQ1MDVlNTYyNjdjMTZiZWQ4MjI4NDI1N2MzYjgxMzI3NTNjYzdmNjRiMTM3YTY4YSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.K1DBTl7H0BLD648J8K9dOaG8TY7AjOFBmolkiG4Ljj0)

Yes this is something really anoying, but I think it is more work. As far as I remember then you have to dig into scintilla
